### DISCUSSION TITLE
Looking for advise on writing tests

### DISCUSSION THREAD

**Discussion by 0xkarambit** [DiveHQ/backend-internship-task#46] [Q&A]
I could not get enough time to code the API so i skipped writing tests for all endpoints, (wrote the whole thing in 2 days only).
But I still want to complete it even though it wouldn't be considered but i just want to do it for the sake of learning.

right now i am writing the test using `fastapi.testclient` and `pytest`,

But writing unit tests for each endpoint considering RBAC seems like a lot of manual work.

Is there a better way to test CRUD APIs with RBAC ? how did y'all write tests for your APIs ?

**Comment by md-abid-hussain** [comment]
That's great. I have a resource which helped me in learning the api development using fastapi.
Tutorial by [Sanjeev-Thiyagarajan](https://www.youtube.com/watch?v=0sOvCWFmrtA&t=51291s)

**Reply by 0xkarambit** [reply]
Thank You bro !
